### PR TITLE
docs(replays): Add blueprint for replay-viewed endpoints

### PR DESCRIPTION
**Pull request by cmanallen**
Add blueprint for replay-viewed endpoints.  Supports fetching viewed metadata and writing new viewed entries.

### CODE REVIEW

File: src/sentry/replays/blueprints/api.md [hunk 1]
```diff
@@ -608,3 +608,107 @@ Retrieve a collection of click events associated with a replay.
     ]
   }
   ```
+
+## Replays Viewed [/organizations/<organization_slug>/replays-viewed/]
+
+- Parameters
+
+  - project (optional, string)
+  - id (required, array[string]) - The id of the replay to retrieve. Can be specified multiple times.
```
Comment (aliu3ntry) [inline src/sentry/replays/blueprints/api.md:617]: Change to "ids" to emphasize this is plural

File: src/sentry/replays/blueprints/api.md [hunk 2]
```diff
@@ -608,3 +608,107 @@ Retrieve a collection of click events associated with a replay.
...
+  - per_page (optional, number) - The per page limit should match the number of ids requested.
```
Comment (aliu3ntry) [inline src/sentry/replays/blueprints/api.md:618]: What's the purpose of this if it always matches the length of the ids array?

File: src/sentry/replays/blueprints/api.md (hunk 1 above)
Comment (aliu3ntry) [reply src/sentry/replays/blueprints/api.md:617]: What's meant by "specify multiple times"?

Comment (aliu3ntry) [reply src/sentry/replays/blueprints/api.md:617]: Is there any difference between this array and the one returned by the endpoint? Does the endpoint maybe validate + check for existence of ids?

File: src/sentry/replays/blueprints/api.md [hunk 3]
```diff
@@ -608,3 +608,107 @@ Retrieve a collection of click events associated with a replay.
...
+    Default: 10
+  - offset (optional, number)
+    Default: 0
+
+### Browse Replays Viewed [GET]
+
+Returns a collection of replays and their viewed state. The user is targetted based on the authorization context. Replay-ids are specified in the URL parameters.
```
Comment (michellewzhang) [inline src/sentry/replays/blueprints/api.md:621]:
```suggestion
Returns a collection of replays and their viewed state. The user is targeted based on the authorization context. Replay-ids are specified in the URL parameters.
```

File: src/sentry/replays/blueprints/api.md [hunk 4]
```diff
@@ -608,3 +608,107 @@ Retrieve a collection of click events associated with a replay.
...
+    "data": [
+      {
+        "id": "81220fbf68494b30a5129ceb4cc5937d",
+        "has_seen": true
+      }
+    ]
+  }
+  ```
+
+## Replay Viewed [/organizations/<organization_slug>/replays/<replay_id>/viewed/]
```
Comment (michellewzhang) [inline src/sentry/replays/blueprints/api.md:649]: Can we title this to specify more clearly that it's an endpoint for fetching the replay _viewers_?

Maybe `Replay Viewers` or `Replay Viewed By`

File: src/sentry/replays/blueprints/api.md (hunk 1 above)
Comment (cmanallen) [reply src/sentry/replays/blueprints/api.md:617]: The parameter is specified like this: `?id=a&id=b`.

Comment (cmanallen) [reply src/sentry/replays/blueprints/api.md:617]: No, no need to validate.  We just want to know if the replay-id exists in the set or not.

File: src/sentry/replays/blueprints/api.md [hunk 5]
```diff
@@ -608,3 +608,107 @@ Retrieve a collection of click events associated with a replay.
...
+- Parameters
+
+  - project (optional, string)
+  - id (required, array[string]) - The id of the replay to retrieve. Can be specified multiple times.
+  - per_page (optional, number) - The per page limit should match the number of ids requested.
+    Default: 10
+  - offset (optional, number)
+    Default: 0
+
+### Browse Replays Viewed [GET]
```
Comment (michellewzhang) [inline src/sentry/replays/blueprints/api.md:619]:
```suggestion
### Browse Replays Viewed Status [GET]
```

or something similar to indicate that we'll get back a status of viewed/not viewed

File: src/sentry/replays/blueprints/api.md (hunk 2 above)
Comment (cmanallen) [reply src/sentry/replays/blueprints/api.md:618]: There is no purpose.  I'll remove it.

File: src/sentry/replays/blueprints/api.md [hunk 6]
```diff
@@ -608,3 +608,107 @@ Retrieve a collection of click events associated with a replay.
...
+        "id": "81220fbf68494b30a5129ceb4cc5937d",
+        "has_seen": true
+      }
+    ]
+  }
+  ```
+
+## Replay Viewed [/organizations/<organization_slug>/replays/<replay_id>/viewed/]
+
+### Fetch Replay Viewed [GET]
```
Comment (michellewzhang) [inline src/sentry/replays/blueprints/api.md:651]:
```suggestion
### Fetch Replay Viewed By [GET]
```

**Comment by cmanallen** [comment]
Ugh, never merging suggestions again.  This happens every time...

**Comment by cmanallen** [comment]
Opening a new PR since I don't want to bother fixing this.

**Pull request closed without merging**